src/error.rs

```diff
@@ -25,6 +35,29 @@ impl From<serde_cbor::Error> for NetError {
     }
 }
 
+impl From<MakerError> for AppError {
+    fn from(value: MakerError) -> Self {
+        AppError::Maker(value)
+    }
+}
+
+impl From<TakerError> for AppError {
+    fn from(value: TakerError) -> Self {
+        AppError::Taker(value)
+    }
+}
+
+impl From<DirectoryServerError> for AppError {
+    fn from(value: DirectoryServerError) -> Self {
+        AppError::DNS(value)
+    }
+}
+
+impl From<std::io::Error> for AppError {
+    fn from(value: std::io::Error) -> Self {
+        AppError::Maker(MakerError::IO(value))
+    }
+}
```

Comment:
Yes as @KnowWhoami said this should not be a `Form` impl. Insrtead at the site where this error is given use the `map_err` to wrap the io error into relevant maker, Taker or DNS error.

Reply:
Understood.